[hxb] Add NativeStackTrace and Exception as dependencies everywhere it's used

kLabz opened the pull request:
Fixes the following:

```haxe
vfs.putContent("Empty.hx", getTemplate("Empty.hx"));
var args = ["-main", "Empty", "-js", "out.js", "--no-output"];
runHaxe(args);

runHaxeJson(args, ServerMethods.Invalidate, {file: new FsPath("/opt/haxe/current/std/StringTools.hx")});

runHaxe(args);
trace(lastResult.stderr);
```

```
Error: Compiler failure: Unexpected BadModule StringTools (Tainted server/invalidate)
Please submit an issue at https://github.com/HaxeFoundation/haxe/issues/new
Attach the following information:
Haxe: 5.0.0-alpha.1+6903e4c03; OS type: unix;
File "src/compiler/server.ml", line 464, characters 112-119
Called from Server.hxb_reader_api_server#resolve_type in file "src/compiler/server.ml", line 436, characters 10-34
Called from HxbReader.hxb_reader#resolve_type in file "src/compiler/hxb/hxbReader.ml", line 182, characters 12-45
Called from HxbReader.hxb_reader#read_clr.(fun) in file "src/compiler/hxb/hxbReader.ml", line 1866, characters 10-44
Called from CamlinternalLazy.do_force_block in file "camlinternalLazy.ml", line 49, characters 17-27
Called from CamlinternalLazy.force_lazy_block in file "camlinternalLazy.ml" (inlined), line 78, characters 27-67
Called from HxbReader.hxb_reader#read_texpr.loop.loop2 in file "src/compiler/hxb/hxbReader.ml", line 1299, characters 14-26
Called from HxbReader.hxb_reader#read_texpr.loop.loop2 in file "src/compiler/hxb/hxbReader.ml", line 1188, characters 15-22
```
(old stack)

---

Obligatory _Rudy TODO checkboxes of doom_ :copyright: : 

* [x] Add test for above
* [x] Fix server test..
* [x] Look for corresponding issues to close / add test for
* [x] Rebase so it's just assertion commit + rest and then merge with rebase

kLabz committed:
Add NativeStackTrace, Exception and ValueException as dependencies everywhere it's used

kLabz commented:
I can't repro that test failure locally.. :/

```
assertations: 1394
successes: 1393
errors: 0
failures: 1
warnings: 0
execution time: 29.875

results: SOME TESTS FAILURES (success: false)
cases.ServerTests
  test10986: FAILURE F
    line: 540, expected true
```

Simn commented:
```
utest/ui/text/PrintReport.hx:52: 
assertations: 1394
successes: 1394
errors: 0
failures: 0
warnings: 0
execution time: 72.288
```

Doesn't repro for me either.

kLabz committed:
Make sure we have meaningful scom when dealing with exception class dependencies

kLabz commented:
Yeah it was only with `-D disable-hxb-cache`

kLabz committed:
insert_save_stacks should not add dependencies, as it's running after save

Shouldn't trigger type loading either.. but I guess if 'haxe.NativeStackTrace.exceptionStack' feature is set then the type is already loaded, so should be fine

kLabz committed:
Avoid some null_pos
